code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -3,21 +3,35 @@
 
 #define FAILED_TO_ADD 1
 
-/obj/item/borg/upgrade/var/vtec_bonus = 0.25
+/obj/item/borg/upgrade/var/vtec_bonus = 0.25 //Define when
 
 /obj/item/borg/upgrade
 	name = "A borg upgrade module."
 	desc = "Protected by FRM."
 	icon = 'icons/obj/module.dmi'
 	icon_state = "cyborg_upgrade"
-	var/locked = 0
+	var/locked = FALSE
 	var/list/required_module = list()
-	var/add_to_mommis = 0
+	var/add_to_mommis = FALSE
 	var/list/modules_to_add = list()
-	var/multi_upgrades = 0
+	var/multi_upgrades = FALSE
 	w_type=RECYK_ELECTRONIC
 
 
+/obj/item/borg/upgrade/proc/locate_component(var/obj/C, var/mob/living/silicon/robot/R, var/mob/living/user)
```

Comment:
Sorry for not getting to this sooner, but you absolutely should not mix the possible return type here.